src/navigation/task/Complete.js

```diff
@@ -53,19 +76,54 @@ const AttachmentItem = ({ base64, onPressDelete }) => {
   )
 }
 
+const FailureReasonPicker = ({ task, httpClient, onValueChange }) => {
+
+  const { data, isSuccess } = useQuery([ 'task', 'failure_reasons', task['@id'] ], async () => {
+    return await httpClient.get(`${task['@id']}/failure_reasons`)
```

Comment:
To be compatible with older servers, when this endpoint returns a 404, the reason picker shouldn't appear.

Reply:
Tried to hide it if 404 is returns but it was looking clunky, so i preferred to add an error message instead
![image](https://github.com/coopcycle/coopcycle-app/assets/5613584/d7c34741-5dc5-4bb6-82c1-de0805649a49)